src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/Rfc3161Accuracy.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.Asn1;
+
+namespace System.Security.Cryptography.Pkcs.Asn1
+{
+    // https://tools.ietf.org/html/rfc3161#section-2.4.2
+    // 
+    // Accuracy ::= SEQUENCE {
+    //       seconds INTEGER              OPTIONAL,
+    //       millis[0] INTEGER(1..999)    OPTIONAL,
+    //       micros[1] INTEGER(1..999)    OPTIONAL }
+    //
+    // And the ASN.1 module starts as
+    // DEFINITIONS IMPLICIT TAGS
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct Rfc3161Accuracy
+    {
+        [OptionalValue]
+        internal int? Seconds;
+
+        [ExpectedTag(0)]
+        [OptionalValue]
+        internal int? Millis;
+
+        [ExpectedTag(1)]
+        [OptionalValue]
+        internal int? Micros;
+
+        internal Rfc3161Accuracy(long totalMicros)
```

Comment:
nit: zero is valid value for accuracy in the spec;

Reply:
And, apparently I had a test for reading zero successfully, but not for writing it.  (It was written by hand-rolling the tokenInfo DER).  Good catch, fixed.